configserver/src/main/java/com/yahoo/vespa/config/server/application/TenantApplications.java

```diff
@@ -38,32 +53,43 @@
  * @author Ulf Lilleengen
  * @author jonmv
  */
-public class TenantApplications {
+public class TenantApplications implements RequestHandler, ReloadHandler, HostValidator<ApplicationId> {
 
     private static final Logger log = Logger.getLogger(TenantApplications.class.getName());
 
     private final Curator curator;
     private final Path applicationsPath;
     private final Path locksPath;
     private final Curator.DirectoryCache directoryCache;
-    private final ReloadHandler reloadHandler;
     private final Executor zkWatcherExecutor;
+    private final Metrics metrics;
+    private final TenantName tenant;
+    private final List<ReloadListener> reloadListeners;
+    private final ConfigResponseFactory responseFactory;
+    private final HostRegistry<ApplicationId> hostRegistry;
+    private final ApplicationMapper applicationMapper = new ApplicationMapper();
+    private final MetricUpdater tenantMetricUpdater;
+    private final Clock clock = Clock.systemUTC();
 
-    private TenantApplications(Curator curator, ReloadHandler reloadHandler, TenantName tenant,
-                               ExecutorService zkCacheExecutor, StripedExecutor<TenantName> zkWatcherExecutor) {
-        this.curator = curator;
+
+    private TenantApplications(TenantName tenant, GlobalComponentRegistry registry) {
```

Comment:
If all the things are directly available for injection, I suggest to do that instead of using the `GlobalComponentRegistry`.

Reply:
Yes, agreed, just need to do it in steps